sdk/eventhub/azure-eventhub/tests/livetest/asynctests/test_send_async.py

```diff
@@ -12,8 +12,116 @@
+        cont_type_ed = "text/plain"
+        corr_id_ed = "corr_id"
+        mess_id_ed = "mess_id"
+        event_data = EventData(body_ed)
+        event_data.content_type = cont_type_ed
+        event_data.correlation_id = corr_id_ed
+        event_data.message_id = mess_id_ed
+
+        batch = await client.create_batch()
+        batch.add(data_message)
+        batch.add(value_message)
+        batch.add(sequence_message)
+        batch.add(event_data)
+        await client.send_batch(batch)
+        await client.send_batch([data_message, value_message, sequence_message, event_data])
+
+    def check_values(event):
+        raw_amqp_message = event.raw_amqp_message
+        if raw_amqp_message.body_type == AmqpMessageBodyType.DATA:
+            if raw_amqp_message.application_properties and raw_amqp_message.application_properties.get(b'body_type') == b'data':
+                body = [data for data in raw_amqp_message.body]
+                assert data_body == body
+                assert event.body_as_str() == "aabbcc"
+                assert raw_amqp_message.delivery_annotations[b'delann_key'] == b'delann_value'
+                assert raw_amqp_message.application_properties[b'body_type'] == b'data'
+            else:
+                assert event.body_as_json() == {'json_key': 'json_val'}
+                assert event.correlation_id == corr_id_ed
+                assert event.message_id == mess_id_ed
+                assert event.content_type == cont_type_ed
+                assert event.body_type == AmqpMessageBodyType.DATA
+        elif raw_amqp_message.body_type == AmqpMessageBodyType.SEQUENCE:
+            body = [sequence for sequence in raw_amqp_message.body]
+            assert [sequence_body] == body
+            assert event.body_as_str() == "['message', 123.456, True]"
+            assert raw_amqp_message.footer[b'footer_key'] == b'footer_value'
+            assert raw_amqp_message.properties.subject == b'sequence'
+            assert raw_amqp_message.application_properties[b'body_type'] == b'sequence'
+        elif raw_amqp_message.body_type == AmqpMessageBodyType.VALUE:
+            assert raw_amqp_message.body == value_body
+            assert event.body_as_str() == "{'key': [-123, 'data', False]}"
+            assert raw_amqp_message.annotations[b'ann_key'] == b'ann_value'
+            assert raw_amqp_message.application_properties[b'body_type'] == b'value'
+
+    async def on_event(partition_context, event):
+        on_event.received.append(event)
+
+    on_event.received = []
+    client = EventHubConsumerClient.from_connection_string(connection_str,
+                                                        consumer_group='$default')
+    async with client:
+        task = asyncio.ensure_future(client.receive(on_event, starting_position="-1"))
+        await asyncio.sleep(15)
+        for event in on_event.received:
+            check_values(event)
+
+    await task
+
+    assert len(on_event.received) == 8
```

Comment:
sorry for not making it clear, I actually like the previous part where we have dictionary for checking count of different message types, it validates that we receive and parse the right type of messages? could we add it back :)

(also the sync one)